Group1

nduvieilh opened the pull request:
# Summary

Migrated dashboard templates and quickstarts for:

1. aws-alb
2. aws-apigateway-apis
3. aws-apigateway-stages
4. aws-appsync
5. aws-athena
6. aws-autoscaling
7. aws-billing-budgets
8. aws-billing-costs
9. aws-cloudfront

Please see comments for details on each dashboard and quickstart changes.

<!-- DON'T DELETE THIS SECTION BELOW IF SUBMITTING A NEW QUICKSTART -->
## Pre merge checklist

<!-- This CHECKLIST SHOULD BE SUBMITTED FULLY COMPLETE WITH THE PR. IF NOT COMPLETE
THE PR REVIEW WILL BE DELAYED -->

- [x] Did you check you NRQL syntax? - Does it work?
- [x] Did you check your dashboard image quality? -  Do they look good?
- [x] Did you check that your alerts actually work?
- [x] Did you include an InstallPlan and Documentation reference?
- [x] Did you check your descriptive content for voice, tone, spelling and grammar errors?
- [x] Did you attach images of your dashboards to the PR so we can see them working?

### Screenshots

Attach images of any visual changes, such as a dashboard here.

**Aws-alb**
![aws-alb1](https://user-images.githubusercontent.com/33464917/190000393-b8956930-dacf-453b-b58f-d9f2fd29f55c.png)

![aws-alb2](https://user-images.githubusercontent.com/33464917/190000519-e14d4410-a977-4c9b-a996-edecff00a6e5.png)

**aws-apigateway-apis**
![aws-apigateway-apis](https://user-images.githubusercontent.com/33464917/190000576-c20dd89c-8c4e-4ade-9095-e5359ad3ebb7.png)

**aws-apigateway-stages**
![aws-apigateway-stages1](https://user-images.githubusercontent.com/33464917/190000621-32e1d170-35f7-45bb-9cd4-05ba87965f9d.png)

![aws-apigateway-stages2](https://user-images.githubusercontent.com/33464917/190000649-68420e2c-571e-43bc-bbb0-0e592d99d57c.png)

**aws-appsync**
![aws-appsync](https://user-images.githubusercontent.com/33464917/190000706-1a8ad4b8-078b-4b9b-aee0-8e33cc7dcf40.png)

**aws-athena**
![aws-athena](https://user-images.githubusercontent.com/33464917/190000732-60759799-b67a-41ab-bc79-3c100a92a6bb.png)

**aws-autoscaling**
![aws-autoscaling1](https://user-images.githubusercontent.com/33464917/190000764-1e08d19b-d71f-4ebd-a546-5d1e5244cc7e.png)

![aws-autoscaling2](https://user-images.githubusercontent.com/33464917/190000778-c9df896a-fb95-44f7-b5e5-68e867fe1e2c.png)

**aws-billing-budgets**
![aws-billing-budgets](https://user-images.githubusercontent.com/33464917/190000884-1389d56b-e14b-4487-8422-139837d6330b.png)

**aws-billing-costs**
![aws-billing-costs](https://user-images.githubusercontent.com/33464917/190001143-5e6535ad-da5f-4b59-9777-692be21d3057.png)

**aws-cloudfront**
![aws-cloudfront](https://user-images.githubusercontent.com/33464917/190000926-14c8443a-b038-444f-b830-c5dcffa2ff3d.png)

Commit:
Added group of dashboards for aws

nduvieilh commented:
# Dashboard 1

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 2

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 3

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 4

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 5

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 6

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 7

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 8

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 9

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

# Dashboard 10

## 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
## 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce commented:
# Dashboard 1 - aws-alb

### Dashboard
For the migrated infra dashboard `/aws-alb/aws-alb.json`, there was already a previously existing `newrelic-quickstarts/dashboards/alb.json`.

This json template seems incorrectly formatted and is missing 2 queries that the migrated template json has. I will wait for what cloud-integrations says, but I believe we can remove the `alb.json` in favor of the `aws-alb.json`

Also, many of IO's existing dashboards include the WHERE clause `WHERE collector.name='cloudwatch-metric-streams'` in many of the queries. This seems to make the query not work as when this is removed, data is returned.

### Quickstart
For the quickstart, there is a `/aws-nlb-alb/config.yml` which points to the `alb` dashboard. Since `newrelic-quickstarts` does not treat alb and nlb as separate templates, I added `aws-alb` and `aws-nlb` to the dashboard property *until* we decide what to do with `alb.json. 

## Dashboard 1 Validation Checklist - aws-alb

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce committed:
feat(group1-validate): Validated aws-alb json

Quickstart aws-alb was created. A logo for the service was added. Validated aws-alb json template.
Added reference screenshots.

lroyce committed:
feat(group1-validate): Removed cloudwatch collector where clause

lroyce committed:
feat(group1-validate-api-gateway-api): Validated dashboard template

Updated dashboard template json with limit property on vizco bullet object.
Compared config for quickstarts, added new api specific directory and added config and logo svg to it.

lroyce committed:
feat(group1-validate): Updated api gateway api json and config with 'by api' details

lroyce committed:
feat(group1-validate-api-gateway-stages): Validated json template

Validated aws api gateway stages template works.
Added subdir under aws/aws-api-gateway-stages/ to which the config yml and logo svg were added. It is the same logo as
was used in aws-api-gateway-apis as the only difference is the queries.
Added screenshots to dashboard directory

lroyce committed:
feat(group1-validate-aws-autoscaling): Validated dashboard template

Fixed a typo in query from regex removal. Added 2 screenshots. Added config and logo to quickstart aws-autoscaling/.
Also, updated installPlans on config to have aws-cloudwatch-metric-streams.

lroyce commented:
## Dashboard 2 - aws-api-gateway-apis

### Dashboard
There was not a previous existing newrelic-quickstart dashboard json for api-gateway - apis. I had to modify the json template as the migration script did not include the `limit` property in `nrqlQueries` which was required to render a vizco bullet widget. 

### Quickstart
 There was a newrelic-quickstart subdir already: `newrelic-quickstart/quickstarts/aws/aws-api-gateway/`. ~~Because of that, I created a separate subdirectory `newrelic-quickstart/quickstarts/aws/aws-api-gateway-apis/` which is more correct as we have two dashboard templates for api-gateway. I made sure the config.yml for api-gateway-apis and api-gateway-stages called out the specific differences between the 2 dashboards.~~ I added the two dashboards `aws-apigateway-apis` and `aws-apigateway-stages` to the already existing io quickstart `aws-api-gateway` config.

# Dashboard 2 Validation Checklist - aws-apigateway-apis

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [x] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce commented:
## Dashboard 3 - aws-api-gateway-stages

### Dashboard
There was not a previous existing newrelic-quickstart dashboard template for api-gateway-stages.

### Quickstart
There was a newrelic-quickstart subdir already at `newrelic-quickstart/quickstarts/aws/aws-api-gateway/`. Because of that, I created a separate subdirectory `newrelic-quickstart/quickstarts/aws/aws-api-gateway-stages/` which is more correct as we have two dashboard templates for api-gateway.

I made sure the config.yml for api-gateway-apis and api-gateway-stages called out the specific differences between the 2 dashboards.

# Dashboard 3 Validation Checklist - aws-apigateway-stages

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [x] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce commented:
## Dashboard 4 - aws-autoscaling
### Dashboard
There was no previously existing dashboard template for api-autoscaling so I created a subdir for the migrated template at `newrelic-quickstart/dashboards/aws-autoscaling/`.
 
I did have to modify 2 queries of the migrated json template as the migration script had removed 'and' from `GroupStandbyInstances`.

### Quickstart
 There was a newrelic-quickstart subdir already: `newrelic-quickstart/quickstarts/aws/aws-auto-scaling/`. Because of that, I created a separate subdirectory `newrelic-quickstart/quickstarts/aws/aws-autoscaling/`.

One difference between the two config.ymls is that the installPlans for the existing aws auto scaling was for `aws-infrastructure-monitoring`. According to the docs, it does say that `aws-cloudwatch-metric-streams` can and should be used.

# Dashboard 4 Validation Checklist - aws-autoscaling

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [x] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce committed:
feat(validate-aws-alb): Updated the screenshots on migrated alb dashboard

lroyce committed:
feat(group1-validation): Updated alb-nlb logo and quickstart config now points to BOTH nlb json and alb json.

lroyce committed:
feat(group1-validate-aws-billing-budgets):

Added aws-billing-budgets-costs subdirectory under quickstarts aws directory, this points to both aws-billing-budgets and aws-billing-costs jsons.
Updated the logo as the existing subdir of aws-billing appears to have an outdated logo.

lroyce commented:
## Dashboard 5 - aws-billing-budgets
### Dashboard

There was a previous existing `newrelic-quickstart/dashboards/aws-billing/` subdir with `aws-billing` json. I created the `newrelic-quickstarts/aws-billing-budgets/` subdir for our json migration.

Differences between existing and migrated template:
Migrated template has a table widget as well as markdown informational widget.

### Quickstart
There was a newrelic-quickstart subdir already: `newrelic-quickstart/quickstarts/aws/aws-billing/`. ~~Because of that, I created a separate subdirectory `newrelic-quickstart/quickstarts/aws/aws-billing-budgets-costs/`. The config.yml I added to this subdir points to both `aws-billing-budgets` and `aws-billing-costs`~~ Updated the already existing config yml dashboard property to point to both `aws-billing-budgets` and `aws-billing-costs`.

# Dashboard 5 - aws-billing-budgets

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce committed:
feat(group1-validate-aws-billing-costs): Validated migrated aws-billing-costs json template.

The new quickstart subdir aws-billing-budget-costs has the quickstart config that points to both aws-billing-budgets and aws-billing-costs jsons.

lroyce commented:
## Dashboard 6 - aws-billing-costs
### Dashboard
There was a previous existing `newrelic-quickstart/dashboards/aws-billing/` subdir with `aws-billing` json. I created the `newrelic-quickstarts/aws-billing-costs/` subdir for our json migration templates `aws-billing-costs.json`.

The existing template aws-billing.json does have the 2 queries that are separated out on the migrated aws-billings-cost.json

Differences between existing and migrated template:
There was no previously existing aws-billing-costs specific json. Though the the queries included on this template were on the aws-billing template that already existed in quickstarts.

### Quickstart
 There was a newrelic-quickstart subdir already: `newrelic-quickstart/quickstarts/aws/aws-billing/`. Because of that, I created a separate subdirectory `newrelic-quickstart/quickstarts/aws/aws-billing-budgets-costs/`. The config.yml I added to this subdir points to both `aws-billing-budgets` and `aws-billing-costs`

# Dashboard 8 Validation Checklist - aws-billing-costs

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce commented:
## Dashboard 7 - aws-appsync
### Dashboard
There was no previous existing quickstart template, so subdir `newrelic-quickstarts/dashboards/aws-appsync/` was created to home aws-appsync.json. 

This template does NOT have a markdown informational widget. We may want to add that. 

### Quickstart
There was a `newrelic-quickstart/aws/aws-appsync` subdir already. I added `aws-appsync` json and dashboard property into config. I updated the installPlans to point to `aws-cloudwatch-metric-streams`. ~~So I created a separate subdirectory until we can compare and decide which one will remain: `newrelic-quickstart/quickstarts/aws/aws-appsync-migrated/`. Differences between existing and migrated config.yml: The config.yml `installPlans` was updated to aws-cloudwatch-metric-streams~~ 


# Dashboard 6 Validation Checklist - aws-appsync

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [x] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce committed:
feat(group1-validate-aws-appsync): Validated aws-appsync template json.

Due to existing quickstart aws subdir for aws-appsync, created aws-appsync-migrated for later comparison between two configs.
Did add updated logo.

lroyce commented:
## Dashboard 8 - aws-athena
### Dashboard
There was a previously existing quickstart template at `newrelic-quickstarts/dashboards/athena/`. The migrated `aws-athena.json` is now located within`dashboards/aws-athena/`. 

This template does NOT have a markdown informational widget. We may want to add that. 

Differences between existing and migrated templates:
All the queries in both templates are different. We will need to compare between existing and migrated templates.
I also updated the titles for the widgets by separating words.

### Quickstart
 There was a `newrelic-quickstart/aws/amazon-athena` subdir already. ~~So I created a separate subdirectory at `newrelic-quickstart/aws/aws-athena` until we can compare and decide which one will remain.~~ Added config dashboard property to point to both `aws-athena` as well as previous existing template `athena`. The config.yml `installPlans` was updated to `aws-cloudwatch-metric-streams`.

# Dashboard 7 Validation Checklist - aws-athena

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [x] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [x] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce committed:
feat(group1-validate-aws-athena): Validated migrated aws-athena.

Separated words in title of widgets. There is an existing amazon-athena template, queries are different.

Also created a aws quickstarts sub, there was an existing subdir so we will need to compare.

lroyce commented:
## Dashboard 8 - aws-cloudfront
There were a number of issues with this migration - the migration script failed and somehow overwrote the file already in place within `newrelic-quickstarts/aws-cloudfront`, I replaced this file and created a new dashboard subdirectory. Theoretically when I push this content, it should not show any differences between this and the main branch.

Another issue is that I could not find an account using the infra-dashboard-templates json to compare against. 

### Dashboard
There was a previously existing quickstart template at `dashboards/aws-cloudfront/`. The migrated subdir `dashboards/aws-cloudfront-migrated/` is home to `aws-cloudfront-migrated.json`. 

Comparing the previously existing `aws-cloudfront` reveals that these two templates are VERY different. 

### Quickstart
 There was a previously existing quickstart config at `newrelic-quickstart/aws/aws-cloudfront`. ~~I created a separate subdirectory at `newrelic-quickstart/aws/aws-cloudfront-migrated` until we can compare and decide which one will remain.~~ I added the dashboard `aws-cloudfront-dashboard` template to the dashboard property of the previously existing config yml. 

Differences between existing and migrated config.yml:
The migrated config.yml `installPlans` was updated to aws-cloudwatch-metric-streams. The description was on a single line, i separated it out so it follows the formatting in other config.ymls.

# Dashboard 9 Validation Checklist - aws-cloudfront

## 1. Validate dashboard
  - [x] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
## 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
## 3. Validate expected differences
  - [x] All widget titles are sentence case
  - [x] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [x] `inventory` widget(s) are removed (if applicable)
  - [x] `event-feed` widget(s) are removed (if applicable)
## 4. Take a screenshot
  - [x] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [x] Add screenshot to dashboard directory as a `*.png` file
## 5. Add config.yml
  - [x] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
## 6. Confirm contents of config.yml
  - [x] Documentation link
  - [x] Appropriate `installPlan`
  - [x] Logo (Preferably a `logo.svg` file in the same directory)
  - [x] Description (can be generic)
  - [x] Reference to the dashboard(s)
## 7. Clean up
  - [x] Remove the `*_validate.json` file from the dashboard directory
  - [x] Remove the `config.yml` file from the dashboard directory
  - [x] Clean up any `*_validate` dashboard that you created in NR1
## 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce committed:
feat(group1-validate-cloudfront): Validated aws-cloudfront

Had nothing to compare migrated template against as no account seemed to be using previous infra template.
Found an account that did have cloudfront data and created dashboard there.
There was both a previous quickstart dashboard and quickstart, created unique subdirs for both at aws-cloudfront-migrated/

lroyce committed:
feat(group1-cloudtrail): Removing aws cloudtrail as json has not been migrated to dimensional metrics yet

lroyce commented:
aws-cloudtrail
Was not included as the template is not migrated to dimensional metrics yet. 

## Dashboard 10 Validation Checklist - aws-cloudtrail

### 1. Validate dashboard
  - [ ] Copy the `*_validate.json` to an account
    - If the accountId does not have data, find an account ID that does, replace the accountId properties with the new acountId
### 2. Compare against infra-dashboard
  - [ ] Find a infra dashboard with data, copy its providerAccountId
  - [ ] Add a providerAccountId to the filter on the migrated dashboard
  - [ ] Document differences
    - If changes are basic, make them directly to the template
    - If changes need to be applied to all dashboards: change script, re-migrate
### 3. Validate expected differences
  - [ ] All widget titles are sentence case
  - [ ] "Dashboard version x.y.z" is removed from markdown (if applicable)
  - [ ] `inventory` widget(s) are removed (if applicable)
  - [ ] `event-feed` widget(s) are removed (if applicable)
### 4. Take a screenshot
  - [ ] Take a screenshot of the dashboard
    - Make sure it does not include any PII
  - [ ] Add screenshot to dashboard directory as a `*.png` file
### 5. Add config.yml
  - [ ] Validate if a `quickstarts/**/*` directory/file exist
    - If not, move the one from the dashboard directory
    - If so, compare it to the one in the dashboard directory; make changes as necessary
### 6. Confirm contents of config.yml
  - [ ] Documentation link
  - [ ] Appropriate `installPlan`
  - [ ] Logo (Preferably a `logo.svg` file in the same directory)
  - [ ] Description (can be generic)
  - [ ] Reference to the dashboard(s)
### 7. Clean up
  - [ ] Remove the `*_validate.json` file from the dashboard directory
  - [ ] Remove the `config.yml` file from the dashboard directory
  - [ ] Clean up any `*_validate` dashboard that you created in NR1
### 8. Move the PR from draft
  - [ ] Create the actual PR, ensure all PR **Pre merge checklist** items are complete

lroyce committed:
fix(group1-validate-cloudfront): Fixing spacing issues so file is same as main branch
